src/gossip/config/generate.lisp

```diff
@@ -3,63 +3,43 @@
 (defparameter *nominal-gossip-port* 65002 "Duplicate of same symbol in gossip package
     to avoid package interdependencies")
 
-#+OBSOLETE
-(defparameter *aws-example* 
-  '((:hostname "emq-01.aws.emotiq.ch" :ip "34.239.111.18" :port 65002)
-    (:hostname "emq-02.aws.emotiq.ch" :ip "52.54.224.84"  :port 65002)
-    (:hostname "emq-03.aws.emotiq.ch" :ip "34.238.39.58"  :port 65002)))
+#+(or)
+(defun generate-network (records &key (root (emotiq/fs:etc/)))
+  "Generate configuration directories for gossip network for RECORDS at directory ROOT
 
-(defparameter *aws-example* 
-  '((:hostname "zt-emq-01.zt.emotiq.ch" :ip "10.178.2.166")
-    (:hostname "zt-emq-02.zt.emotiq.ch" :ip "10.178.0.71" )
-    (:hostname "zt-emq-03.zt.emotiq.ch" :ip "10.178.15.71")))
+Returns a list of directions created with valid configurations.
 
-(defparameter *max-stake* (truncate 1E9))
-
-(defun generate (records &key (root #p"/var/tmp/conf/") (max-stake *max-stake*))
-  "Generate configuration directories of an Emotiq testnet for RECORDS at directory ROOT
-The ROOT defaults to '/var/tmp/conf'."
+  The ROOT defaults to the value returned by 'EMOTIQ/FS:ETC/'."
   (let* ((key-records (generate-keys records))
          (pubkeys (loop :for record :in key-records :collect (car (fourth record))))
-         (stakes (generate-stakes pubkeys max-stake))
          directories)
     (dolist (key-record key-records)
       (destructuring-bind (host eripa port (public private))
           key-record
         (declare (ignore private))
-        (let ((directory (make-pathname :directory (append (pathname-directory root)
-                                                           (list (format nil "~a:~a" host port)))
-                                        :defaults root)))
-          (emotiq:note "~&Writing configuration to '~a'.~&" directory)
-          (ensure-directories-exist directory)
-          (write-local-machine-conf
-           (merge-pathnames *machine-filename* directory)
-           eripa port public)
-          (write-hosts-conf
-           (merge-pathnames *hosts-filename* directory)
-           key-records)
-          (write-keypairs-conf
-           (merge-pathnames *keypairs-filename* directory)
-           key-records)
-          (write-stakes-conf
-           (merge-pathnames *stakes-filename* directory)
-           stakes)
-          (push directory directories))))
+        (let ((path (make-pathname :directory (append (pathname-directory root)
```

Comment:
Pathname-directory is not guaranteed to return a list.
```
CLHS: X3J13 voted in June 1989 (PATHNAME-SUBDIRECTORY-LIST)   to specify that pathname-directory always returns nil, :unspecific, or a list
```

Reply:
This code has been removed, as `gossip/config` no longer generates networks, but rather information for a node.